File: C++/Easy/Accepted/largest-number-at-least-twice-of-others_1.cpp
class Solution {
public:
    int dominantIndex(vector<int>& nums) {
        
        int max = 0;
        int maxIndex = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] > max) {
                max = nums[i];
                maxIndex = i;
            }
        }
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != max && max < 2 * nums[i]) {
                return -1;
            }
        }
        return maxIndex;
    }
};

File: Rust/Easy/Accepted/largest-number-at-least-twice-of-others_1.rs
impl Solution {
    pub fn dominant_index(nums: Vec<i32>) -> i32 {
        
        let mut max = 0;
        let mut idx = 0;
        for i in 0..nums.len() {
            if nums[i] > max {
                max = nums[i];
                idx = i;
            }
        }
        for i in 0..nums.len() {
            if i != idx && max < 2 * nums[i] {
                return -1;
            }
        }
        idx as i32
    }
}